src/main/java/it/rate/webapp/services/GoogleImageService.java

```diff
@@ -22,14 +23,16 @@ public class GoogleImageService implements ImageService {
   @Value("${UPLOAD_DIRECTORY}")
   private String UPLOAD_DIRECTORY;
 
-  private Drive driveService;
+  private final Drive driveService;
+  private final InterestRepository interestRepository;
 
-  public GoogleImageService(Drive driveService) {
+  public GoogleImageService(Drive driveService, InterestRepository interestRepository) {
     this.driveService = driveService;
+    this.interestRepository = interestRepository;
   }
 
   @Override
-  public String savePlaceImage(MultipartFile image, Long placeId) throws IOException {
+  public String saveImage(MultipartFile image) throws IOException {
 
     FileContent mediaContent;
     String currentUser = SecurityContextHolder.getContext().getAuthentication().getName();

```

Comment:
This is relict from another pull request but for sake of consistency this method could be receiving AppUser obtained from Principal in the Controller instead of getting it from SecurityContext here. This can potentially fail if this method is called from wrong place.